subsys/net/ip/Kconfig

```diff
@@ -10,11 +10,11 @@ menu "IP stack"
 	depends on NETWORKING
 
 # Hidden option
-config NET_L2_RAW_CHANNEL
+config NET_RAW_MODE
```

Comment:
This config option has now become totally obscure as now even the name does not give any hint what it means. As there is no help option one cannot easily figure out what this means. I feel the name of the option is a bit too generic, but I do not know how we should call this.
Biggest issue is the missing help text or proper comment in Kconfig file, we really need to explain what this option is doing.
The rest of the PR looks fine.

Reply:
true, some doc would be indeed necessary here

Reply:
+1 to @jukkar , maybe L2_RAW otherwise NET_RAW seems too vague.

Reply:
No L2_RAW is semantically wrong. It's the full NET stack which basically is avoided. Only the very basic primitives are built-in (net_pkt so far...) So it's really about RAW networking: net drivers get to rx/tx net_pkt, but the whole logic to process what's inside net_pkt is not here.